stable/external-dns/templates/clusterrole.yaml

```diff
@@ -32,19 +32,35 @@ rules:
   - get
   - list
   - watch
+{{- if or .Values.crd.create .Values.crd.apiversion }}
 - apiGroups:
+  {{- if .Values.crd.create }}
   - externaldns.k8s.io
+  {{- else }}
+  - {{ $api := splitn "/" 2 .Values.crd.apiversion }}{{ $api._0 }}
+  {{- end }}
   resources:
+  {{- if .Values.crd.create }}
   - dnsendpoints
+  {{- else }}
+  - {{ printf "%ss" (.Values.crd.kind | lower) }}
+  {{- end }}
   verbs:
   - get
   - list
   - watch
-{{- if .Values.crd.create }}
 - apiGroups:
+  {{- if .Values.crd.create }}
   - externaldns.k8s.io
+  {{- else }}
+  - {{ $api := splitn "/" 2 .Values.crd.apiversion }}{{ $api._0 }}
+  {{- end }}
   resources:
+  {{- if .Values.crd.create }}
   - dnsendpoints/status
+  {{- else }}
+  - {{ printf "%ss/status" (.Values.crd.kind | lower) }}
```

Comment:
What's the difference between %s and %ss?

Reply:
You have to fill in the plural name, we only got the singular, which is probably always "DNSEndpoint" - I haven't opted for an additional values option, since the formal API source has clearly that name.

Reply:
I think it's also possible to generate a CRD with a Kind of a completely different name than plural+group, but man, this is off standard then...